content/doc/developer/cli/writing-cli-commands.adoc

```diff
@@ -0,0 +1,65 @@
+---
+title: Writing CLI commannds
+layout: developersection
+---
+
+Plugins can contribute additional commands to link:/doc/book/managing/cli/[Jenkins CLI] which.
+
+This is useful for:
+
+* exposing administrative commands to admins, so that they can script some of the Jenkins babysitting work, 
+* exposing data and operations to builds executing inside Jenkins, so that they can interact with Jenkins in a richer way.
+
+Writing commands can be done in two ways.
+
+## Annotating a method with CLIMethod
+
+If you are exposing behaviors of your model objects as CLI commands, the easiest way to achieve it is to put link:http://javadoc.jenkins-ci.org/byShortName/CLIMethod[@CLIMethod] on a method of your model object. See link:https://javadoc.jenkins-ci.org/hudson/model/Queue.html#clear--[Queue.clear()] as an example. 
```

Comment:
Please try to use the `jenkinsdoc:` macro for Javadoc links: https://github.com/jenkins-infra/jenkins.io/blob/master/STYLEGUIDE.adoc#javadoc-links

(Several more occurrences in this doc)

Reply:
Does this work for linking to methods?
i.e. I've tried the examples and I can't get it to work...

```
jenkinsdoc:hudson.cli.CLICommand#getShortDescription--
```
links to:
https://javadoc.jenkins.io/hudson/cli/CLICommand#getShortDescription--.html#getShortDescription--

it should link to:
https://javadoc.jenkins.io/hudson/cli/CLICommand.html#getShortDescription--

I can only get it to work for linking to the whole class which isn't really what I wannt...

Reply:
I've refactored for a few but for ~half of the links it didn't work...

Reply:
Looks like this has been broken since @rtyler merged https://github.com/jenkins-infra/asciidoctor-jenkins-extensions/pull/1 despite it being WIP.

I expect the culprit is `simple_class` in https://github.com/jenkins-infra/asciidoctor-jenkins-extensions/blob/d6969670f894baccf5ea97c25d394c23c0e98d77/lib/asciidoctor/jenkins/extensions/javadoc-link.rb#L57 never getting the `gsub(/#.*/, '')` treatment. Cannot get the dev console to work unfortunately.

Reply:
@oleg-nenashev @daniel-beck 
https://github.com/jenkins-infra/asciidoctor-jenkins-extensions/pull/9

Dev console started for me fine, but no idea how to actually use it, but I wrote some tests and fixed it that way, and then verified all the examples worked using the `shotgun` web server thing.